flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -372,10 +374,10 @@ public void unregister(Metric metric, String metricName, AbstractMetricGroup gro
 				if (reporters != null) {
 					for (int i = 0; i < reporters.size(); i++) {
 						try {
-						MetricReporter reporter = reporters.get(i);
-							if (reporter != null) {
-								FrontMetricGroup front = new FrontMetricGroup<AbstractMetricGroup<?>>(i, group);
-								reporter.notifyOfRemovedMetric(metric, metricName, front);
+						ReporterAndSettings reporterAndSettings = reporters.get(i);
+							if (reporterAndSettings != null) {
+								FrontMetricGroup front = new FrontMetricGroup<AbstractMetricGroup<?>>(reporterAndSettings.settings, group);
```

Comment:
access to fields should be consistent in this class. I would suggest to go with `getSettings` as this method was introduced.